Source/Csla.Blazor/Authentication/CslaAuthenticationStateProvider.cs

```diff
@@ -10,6 +10,7 @@ namespace Csla.Blazor.Client.Authentication
   public class CslaAuthenticationStateProvider : AuthenticationStateProvider
```

Comment:
@TheCakeMonster you were wondering if we should even have the CSLA authenticationstateprovider and user service.

I think we do need them, b/c _some_ `AuthenticationStateProvider` must be registered for CSLA to initialize at all (app context requires the service). If I just create a Blazor app with no authn at all, I don't believe any provider is registered, which would prevent your app from running, due to a missing required service.

So at the very least, I think we need to provide simplistic service implementations that support the bare minimum Blazor project templates.

Reply:
Your original comment:

> I have slightly mixed feelings about the presence of CslaAuthenticationStateProvider and CslaUserService. Their presence does support simpler demo code, but do not offer the level of detail of the built in state providers that come in the box with Blazor. It also starts to project Csla into the security space in a way that isn't necessary, putting the framework at risk of containing vulnerabilities.
> When people create a solution using either of the Blazor templates, they already get a state provider, so it might even be more confusing to have Csla versions available. People who don't know this security model well might think that they need to make use of these implementations, when that is not the case.
> The benefit of the Blazor ApplicationContextManager is that it can work with any state provider; as long as it responds to the AuthenticationStateChanged event, then the ones that are provided by Microsoft are perfectly appropriate.
> The only thing I am not absolutely sure of is whether the steps detailed in the book to create the multi-headed solution creates the state providers; I guess this is based on project templates and not solution templates. I know for sure that creating standalone solutions for either Blazor Server or Blazor WebAssembly definitely does create the state providers.

You say here that you are confident that the minimal Blazor project templates (w/o any authn) do register some state provider. If that's the case, then I tend to agree with you that we should use the "dummy" Microsoft ones rather than providing a duplicate "dummy" implementation.

Reply:
I just did a quick test. It appears that some `AuthenticationStateProvider` is registered and available. It contains a null `AuthenticationState` reference.

So I think you are right, we can remove the CSLA implementations of these types and work with the default implementation.